### PR TITLE
Issue 1017 by pjohans: handle errors from opensearch

### PR DESCRIPTION
**Pull request by pjohans**
opensearch delivers a <result> or <error>. Handle the error

### CODE REVIEW

**Comment by pjohans** [comment]
this one works with
https://github.com/ding2/ting-client/pull/10

File: plugins/content_types/search_result.inc [hunk 1]
```diff
@@ -83,17 +83,35 @@ function ting_search_search_result_content_type_render($subtype, $conf, $panel_a
...
+    if (isset($search_result) && in_array($search_result->sortUsed, array_keys($frequency_rank))) {
+      $msg = t('Records are sorted by %sortUsed . Select another sort if it is not suited', array('%sortUsed' => $frequency_rank[$search_result->sortUsed]));
+      $message .= '<div class="messages status">' . $msg . '</div>';
+    }
+
     if (isset($search_result) && $limit < $search_result->numTotalObjects) {
       $msg = variable_get('ting_search_result_message', 'Your search gave more than %s results. Try to search more specific or use the facets to filter the result.');
       $msg = sprintf($msg, $limit);
-      $message = '<div class="messages search-field-in-content--message">' . $msg . '</div>';
+      $message .= '<div class="messages search-field-in-content--message">' . $msg . '</div>';
```
Comment (kasperg) [inline plugins/content_types/search_result.inc]: This seems to be unrelated to the current issue?

File: plugins/content_types/search_result_count.inc [hunk 2]
```diff
@@ -21,7 +21,16 @@ function ting_search_result_count_content_type_render($subtype, $conf, $panel_ar
   $block = new stdClass();
 
   $search_result = drupal_static('ting_search_results');
-  if (isset($search_result)) {
+
+  if(isset($search_result->error)) {
+    $txt = t('An error occurred: %error_text ', array('%error_text' => $search_result->error));
```
Comment (kasperg) [inline plugins/content_types/search_result_count.inc]: I do not think relaying the error message to the end users is a particulary good idea.

1. They will probably not understand it
2. It is probably not something they can do anything about
3. It will not be translated to the site language.

Consequently I think we should leave it out.

File: plugins/content_types/search_result.inc [hunk 3]
```diff
@@ -83,17 +83,35 @@ function ting_search_search_result_content_type_render($subtype, $conf, $panel_a
...
     $block->content = $message . theme('ting_search_results', array('results' => $results['#results'], 'module' => $conf['type']));;
   }
   else {
-    if (empty($conf['override_empty'])) {
+    $search_result = drupal_static('ting_search_results');
+    if(isset($search_result->error)){
+      $block->title = t('An error occured: %error_text ',array('%error_text' => $search_result->error));
+      $block->content = t('<ul><li>Please try again.</li> <li>Contact administrator if problem persists.</li>');
+      $support_url = l(t('Customer service'), 'http://kundeservice.dbc.dk/',array('attributes' => array('target'=>'_blank')));
+      $block->content .= t('<li>You can check service status here: '). $support_url .'</li></ul>';
```
Comment (kasperg) [inline plugins/content_types/search_result.inc]: The relationship between text and markup in the translation is convoluted here. The `<ul><li>` is part of a translatable string while other parts are not.

In general I do not think an unordered list is a suitable choice for this sort of content. Why not use normal `<p>` tags and let the content be translatable?

**Comment by pjohans** [comment]
this was supposed to work with https://github.com/ding2/ting-client/pull/10  but since searchResult->error is never returned from the search as an exception is thrown this pull request is dead

**Comment by kasperg** [comment]
OK. I will close this then.
